src/main/scala/com/fulcrumgenomics/vcf/api/Genotype.scala

```diff
@@ -0,0 +1,91 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.vcf.api
+
+import com.fulcrumgenomics.vcf.api.Allele.{NoCallAllele, SpannedAllele}
+
+/** A genotype for a variant.
+  *
+  * @param alleles the set of alleles for the variant
+  * @param sample the name of the sample for which this is a genotype
+  * @param calls the called alleles for the sample
+  * @param phased whether or not the calls are phased
+  */
+final case class Genotype(alleles: AlleleSet,
+                          sample: String,
+                          calls: IndexedSeq[Allele],
+                          phased: Boolean = false,
+                          attrs: Map[String, Any] = Variant.EmptyGtAttrs
+                         ) {
+  require(calls.nonEmpty, "Genotype must have ploidy of at least 1!.")
+
+  /** The indices of the calls within the AlleleSet. If any allele is no-called, that is returned as -1. */
+  val callIndices: IndexedSeq[Int] = calls.map(c => if (c == NoCallAllele) -1 else alleles.indexOf(c))
+
+  /** Returns the separator that should be used between alleles when displaying the genotype. */
+  private def separator: String = if (phased) "|" else "/"
+
+  /** Returns the set of alleles for the genotype filtering out no-calls and spanned alleles. */
+  private def calledAlleles: Seq[Allele] = calls.filter(a => a != NoCallAllele && a != SpannedAllele)
+
+  /** The ploidy of the called genotype - equal to calls.length. */
+  def ploidy: Int = calls.length
+
+  /** True if all [[calls]] are no-call alleles. */
+  def isNoCall: Boolean = calls.forall(_ == Allele.NoCallAllele)
+
+  /** True if none of [[calls]] are no-call alleles. */
+  def isFullyCalled: Boolean = !calls.contains(Allele.NoCallAllele)
+
+  /** True if all [[calls]] are the reference allele. */
+  def isHomRef: Boolean = calls.forall(_ == alleles.ref)
+
+  /** True if all [[calls]] are the same non-reference allele. */
+  def isHomVar: Boolean = isFullyCalled && calls(0) != alleles.ref && calls.forall(_ == calls(0))
+
+  /** True if the genotype contains at least two called alleles that are different from one another. */
```

Comment:
I _think_ this definition makes most sense given ploidy can be any value, but take a look at the tests and see if you agree.  Same for isHetNonRef.